### PR TITLE
To avoid Fatal error with JEditor::getButtons (for issue #660)

### PR DESCRIPTION
**Pull request by Buddhima**
File: joomla-platform / libraries / joomla / html / editor.php
Method: getButtons()

Issue: If the $className would not exists (due to plugin class is missing) , assigning to $temp variable becomes unsafe

Solution: Put the assiging inside a try catch block, so that not existance of plugin class would not disturb the flow of foreach loop 

Signed-off-by: Buddhima Wijeweera

### CODE REVIEW

**Comment by mbabker** [comment]
You need to fix your code style first - http://developer.joomla.org/pulls/pulls/1093.html

Also, instead of echoing the message, you should be returning an error.  Instead of the try/catch method you've proposed, maybe the current if statement would work fine, but add an else so that an Exception is thrown and the caller can handle the error.

File: libraries/joomla/html/editor.php [hunk 1]
```diff
@@ -455,11 +455,18 @@ public function getButtons($editor, $buttons = true)
 				$plugin = new $className($this, (array) $plugin);
 			}
 
+                        try{
 			// Try to authenticate
-			if ($temp = $plugin->onDisplay($editor, $this->asset, $this->author))
-			{
-				$result[] = $temp;
-			}
+                            // Unsafe opearation, might cause an exception if $className not exists
```
Comment (elinw) [inline libraries/joomla/html/editor.php]: operation

Comment (Buddhima) [inline libraries/joomla/html/editor.php]: :D a mistake. Thanks for pointing out
Corrected !!!

File: libraries/joomla/html/editor.php [hunk 2]
```diff
@@ -426,6 +426,8 @@ public function setContent($editor, $html)
 	 *
 	 * @return  array
 	 *
+         * @throws	InvalidArgumentException	If plugin required does not exists
+         *
```
Comment (chdemko) [inline libraries/joomla/html/editor.php]: Please indent correctly

File: libraries/joomla/html/editor.php [hunk 3]
```diff
@@ -454,12 +456,17 @@ public function getButtons($editor, $buttons = true)
...
+			{
+				throw new InvalidArgumentException('Class does not exists: ' . $className);
+			}
 
 			// Try to authenticate
 			if ($temp = $plugin->onDisplay($editor, $this->asset, $this->author))
 			{
 				$result[] = $temp;
 			}
+
```
Comment (chdemko) [inline libraries/joomla/html/editor.php]: superfluous line

**Comment by chdemko** [comment]
The pull request cannot be merged

**Comment by Buddhima** [comment]
Hi, can you tell me why it is failing?

**Comment by chdemko** [comment]
you have to merge with the current trunk

**Comment by elinw** [comment]
The issue is not that the pull request can't be merged (that would be shown on the pull tester report with red)  All of the pull requests with ?? in them have the same failure. I'm not sure if it is because they all have a dependency on JHtml or not, but that is possible.  Those test reports were generated on May 5 and have not been updated since.
All of them show this as the failing line
https://github.com/joomla/joomla-platform/blob/staging/tests/suites/unit/joomla/html/JHtmlTest.php#L896

**Comment by Buddhima** [comment]
Thanks Elin for pointing to the correct place :)

**Comment by elinw** [comment]
Okay now you need to update your branch, since the pulltester has rerun and it says it is not able to merge cleanly.

**Comment by realityking** [comment]
This still doesn't merge cleanly. Also I'm wondering under what circumstances this would be triggered.

**Comment by elinw** [comment]
@realityking  if the plugin is missing but you put the button in the field definition, I believe.

@Buddhima will you please rebase this or close it until you get it rebased?

**Comment by LouisLandry** [comment]
@Buddhima the library in question has been removed from the platform and added to the CMS tree.  I'd suggest you submitting this to the CMS repository since it is a very smart and correct change in behavior.  I'm closing this since it is no longer relevant for the platform.  Apologies for not getting to it sooner.

**Pull request closed without merging**